modules/migration/comment.go

```diff
@@ -18,13 +18,15 @@ type Commentable interface {
 type Comment struct {
 	IssueIndex  int64 `yaml:"issue_index"`
 	Index       int64
+	CommentType int64  `yaml:"comment_type"`
 	PosterID    int64  `yaml:"poster_id"`
 	PosterName  string `yaml:"poster_name"`
 	PosterEmail string `yaml:"poster_email"`
 	Created     time.Time
 	Updated     time.Time
 	Content     string
 	Reactions   []*Reaction
+	Meta        map[string]interface{} `yaml:"meta,omitempty"` // see models/issues/comment.go for fields in Comment struct
```

Comment:
How does the value of Meta gets updated? I can only see some code reading it, but I can not find code writing it.

If let me decide, I would use clear field names instead of the dynamic map: the field names are clearly known, the dynamic map make the code like a dynamic language which loses the static compiling time check

Reply:
It is written to via custom tooling to comments.yml and used during "restore-repo" CLI command. 

I used an interface because this in theory could be extended to fit each field in the comments struct and I can import that struct due to cyclical imports.

Reply:
> It is written to via custom tooling to comments.yml and used during "restore-repo" CLI command.

Is there a related PR to show how to use it?

> I used an interface because this in theory could be extended to fit each field in the comments struct and I can import that struct due to cyclical imports.

I do not think the "in theory could be extended to fit each field" could really work with a dynamic map. In the end, on the reader side, every field is still pre-defined and clearly hard-coded in code. If it is the case, why not put some pointers like: `AssigneeID *uint64`, `OldTitle *string`.